Riot/Modules/Room/DataSources/RoomDataSource.m

```diff
@@ -963,4 +974,72 @@ - (void)bubbleReactionsViewModel:(BubbleReactionsViewModel *)viewModel didLongPr
     [self.delegate dataSource:self didRecognizeAction:kMXKRoomBubbleCellLongPressOnReactionView inCell:nil userInfo:@{ kMXKRoomBubbleCellEventIdKey: eventId }];
 }
 
+#pragma mark - Room creation intro cell
+
+- (BOOL)canShowRoomCreationIntroCell
+{
+    NSString* userId = self.mxSession.myUser.userId;
+
+    if (!userId || !self.isLive || self.isPeeking)
+    {
+        return NO;
+    }
+    
+    // Room creation cell is only shown for the creator
+    return [self.room.summary.creatorUserId isEqualToString:userId];
+}
+
+- (void)enableRoomCreationIntroCellDisplayIfNeeded
+{
+    self.showRoomCreationCell = [self canShowRoomCreationIntroCell];
+}
+
+// Insert the room creation intro cell at the begining
+- (void)insertRoomCreationIntroCellDataIfNeeded
+{
+    if (self.showRoomCreationCell)
+    {
+        NSUInteger existingRoomCreationCellDataIndex = [self roomBubbleDataIndexWithTag:RoomBubbleCellDataTagRoomCreationIntro];
+        
+        if (existingRoomCreationCellDataIndex != NSNotFound)
+        {
+            [bubbles removeObjectAtIndex:existingRoomCreationCellDataIndex];
+        }
+        
+        NSUInteger roomCreationConfigCellDataIndex = [self roomBubbleDataIndexWithTag:RoomBubbleCellDataTagRoomCreateConfiguration];
+        
+        // Only add room creation intro cell if `bubbles` array contains the room creation event
+        if (roomCreationConfigCellDataIndex != NSNotFound)
+        {
+            MXEvent *event = [MXEvent new];
+            MXRoomState *roomState = [MXRoomState new];
+            RoomBubbleCellData *roomBubbleCellData = [[RoomBubbleCellData alloc] initWithEvent:event andRoomState:roomState andRoomDataSource:self];
+            roomBubbleCellData.tag = RoomBubbleCellDataTagRoomCreationIntro;
+            
+            [bubbles insertObject:roomBubbleCellData atIndex:0];
+                
+            self.roomCreationCellData = roomBubbleCellData;
+        }
+    }
+    else
+    {
+        self.roomCreationCellData = nil;
+    }
+}
+
+- (NSUInteger)roomBubbleDataIndexWithTag:(RoomBubbleCellDataTag)tag
+{
+    return [bubbles indexOfObjectPassingTest:^BOOL(id<MXKRoomBubbleCellDataStoring>  _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
```

Comment:
Similar methods in `MXKRoomDataSource` are protected with `@synchronized(bubbles)`